src/angular2Apollo.ts

```diff
@@ -23,12 +36,29 @@ export const defaultApolloClient = (client: ApolloClient): Provider => {
 export class Angular2Apollo {
   constructor(
     @Inject(angularApolloClient) private client: any
-  ) {
-
-  }
+  ) {}
 
   public watchQuery(options): ApolloQueryObservable<any> {
-    return new ApolloQueryObservable(this.client.watchQuery(options));
+    const apolloRef = new ObservableQueryRef();
+    if (typeof options.variables === 'object') {
+      const varObs = observeVariables(options.variables);
+
+      return new ApolloQueryObservable(apolloRef, subscriber => {
+        const sub = varObs.switchMap(newVariables => {
+          const cleanOptions = omit(options, 'variables');
+          const newOptions = assign(cleanOptions, { variables: newVariables });
+
+          apolloRef.apollo = this.client.watchQuery(newOptions);
```

Comment:
@kamilkisiela would it be better to do a refetch with the new variables instead of creating a new query here?